pennylane_runtime/vqe_runtime.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module contains a custom VQE runtime program that can be uploaded to IBMQ.
+"""
+# pylint: # pylint: disable=too-many-arguments,too-many-branches,too-many-statements
+
+import numpy as np
+import scipy.optimize as opt
+from scipy.optimize import OptimizeResult
+import mthree
+
+import qiskit.circuit.library.n_local as lib_local
+from qiskit.algorithms.optimizers import SPSA, QNSPSA
+from qiskit import QuantumCircuit, transpile
+
+
+def opstr_to_meas_circ(op_str):
+    """Takes a list of operator strings and makes circuit with the correct post-rotations for measurements.
+
+    Args:
+        op_str (list): List of strings representing the operators needed for measurements.
+
+    Returns:
+        list: List of circuits for rotations before measurement.
+    """
+    num_qubits = len(op_str[0])
+    circs = []
+    for op in op_str:
+        qc = QuantumCircuit(num_qubits)
+        for idx, item in enumerate(op):
+            if item == "X":
+                qc.h(idx)
+            elif item == "Y":
+                qc.sdg(idx)
+                qc.h(idx)
+            elif item == "H":
+                qc.ry(-np.pi / 4, idx)
+        circs.append(qc)
+    return circs
+
+
+def main(
+    backend,
+    user_messenger,
+    hamiltonian,
+    x0,
+    ansatz="EfficientSU2",
+    ansatz_config=None,
+    optimizer="SPSA",
+    optimizer_config=None,
+    shots=8192,
+    use_measurement_mitigation=False,
+):
+    """
+    The main sample VQE program.
+
+    Parameters:
+        backend (ProgramBackend): Qiskit backend instance.
+        user_messenger (UserMessenger): Used to communicate with the
```

Comment:
Good to provide the absolute import paths for `ProgramBackend` and `UserMessenger`, e.g., `Qiskit.IBMQ` instead of `IBMQ`.